src/Components/DynamicFields.res

```diff
@@ -480,218 +277,394 @@ let make = (
     }
   }
 
+  let dynamicFieldsToRenderOutsideBilling =
+    fieldsArr->Js.Array2.filter(field =>
+      PaymentMethodsRecord.dynamicFieldsToRenderOutsideBilling->Js.Array2.includes(field)
+    )
+
+  let dynamicFieldsToRenderInsideBilling =
+    fieldsArr->Js.Array2.filter(field =>
+      !(PaymentMethodsRecord.dynamicFieldsToRenderOutsideBilling->Js.Array2.includes(field))
+    )
+
+  let isOnlyInfoElementPresent =
+    dynamicFieldsToRenderInsideBilling->Js.Array2.length === 1 &&
+      dynamicFieldsToRenderInsideBilling->Belt.Array.get(0) === Some(InfoElement)
+
+  let isRenderDynamicFieldsInsideBilling =
+    dynamicFieldsToRenderInsideBilling->Js.Array2.length > 1 || !isOnlyInfoElementPresent
+
```

Comment:
The above has to handle the zero fields usecase for when dynamic fields required is none.